grails-app/services/com/unifina/service/CommunityProductService.groovy

```diff
@@ -0,0 +1,89 @@
+package com.unifina.service
+
+import com.unifina.api.ApiException
+import com.unifina.api.CommunityJoinRequestCommand
+import com.unifina.api.NotFoundException
+import com.unifina.api.UpdateCommunityJoinRequestCommand
+import com.unifina.domain.community.CommunityJoinRequest
+import com.unifina.domain.community.CommunitySecret
+import com.unifina.domain.security.IntegrationKey
+import com.unifina.domain.security.SecUser
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
+import org.springframework.security.crypto.password.PasswordEncoder
+
+class CommunityProductService {
+	PasswordEncoder encoder = new BCryptPasswordEncoder()
+
+	void onApproveJoinRequest(CommunityJoinRequest c) {
+		c.state = CommunityJoinRequest.State.ACCEPTED
+		// Fetch joinPartStream id from smart contract at address
+		// Backend produces join message to joinPartStream
+	}
+
+	List<CommunityJoinRequest> findCommunityJoinRequests(String communityAddress, CommunityJoinRequest.State st) {
+		if (st != null) {
```

Comment:
Criteria query would allow you to write the conditional requirements in a cleaner way:
```
CommunityJoinRequest.withCriteria {
    eq("communityAddress", communityAddress")
    if (st) {
        eq("state", st.toString())
    }
}
```